File: Nov2020/26Nov/PrintInReverse.cpp
/*
Problem name: Print in reverse
Problem link: https://www.hackerrank.com/challenges/print-the-elements-of-a-linked-list-in-reverse/problem
*/

/*
 * For your reference:
 *
 * SinglyLinkedListNode {
 *     int data;
 *     SinglyLinkedListNode* next;
 * };
 *
 */
void reversePrint(SinglyLinkedListNode* head) {
    SinglyLinkedListNode* temp = head;
    stack<int> s;
    while(temp){
        s.push(temp->data);
        temp=temp->next;
    }
    while(!s.empty()){
        cout<<s.top()<<"\n";
        s.pop();
    }

}

File: Nov2020/26Nov/PrintInReverse.java
/*
Problem Name: Print in Reverse
Problem Link: https://www.hackerrank.com/challenges/print-the-elements-of-a-linked-list-in-reverse/problem
*/
// This file contains only the function which needs to be edited

/*
* For your reference:
*
* SinglyLinkedListNode {
*     int data;
*     SinglyLinkedListNode next;
* }
*
*/
static void reversePrint(SinglyLinkedListNode head) {
    if(head == null)
        return;
    reversePrint(head.next);
    System.out.println(head.data);
}
